activerecord/test/cases/nested_attributes_test.rb

```diff
@@ -74,6 +75,7 @@ def test_should_raise_an_UnknownAttributeError_for_non_existing_nested_attribute
   end
 
   def test_should_disable_allow_destroy_by_default
+    Pirate.nested_attributes_options.delete :ship
```

Comment:
Can be this be done in `setup` or is it not needed in every test?

Reply:
The issue is that some tests relying on the existing declaration, so they don't redeclare inside the test body, I thought this was the easiest way to get these passing again, but I'm also happy to update them so they're all consistent, so we can just do this in the setup.

Reply:
You can move the ones that need this to a new test class so that you can do it in setup.

Reply:
@eileencodes

These are existing tests that are already in specific classes or modules, would it make sense to move them to another (nested?) class just for this?

Also, the association being asserted will vary with each test. Are you suggesting a nested class for each association? Otherwise removing the declaration for an association that the test doesn't rely on may have side effects, I'd rather be explicit about it.